packages/jest-cli/src/watch.js

```diff
@@ -269,12 +307,10 @@ export default function watch(
         startRun(globalConfig);
         break;
       case KEYS.C:
-        globalConfig = updateGlobalConfig(globalConfig, {
-          mode: 'watch',
+        updateConfigAndRun({
           testNamePattern: '',
           testPathPattern: '',
         });
-        startRun(globalConfig);
         break;
       case KEYS.F:
```

Comment:
should this be a plugin as well?

Reply:
Correct, this one should be a plugin... I can do it in this PR of as follow up PRs

Reply:
I also want to wait because it will require some refactor